src/services/Carts.php

```diff
@@ -537,4 +538,27 @@ private function _getCartPaymentCurrencyIso(): string
 
         return Plugin::getInstance()->getPaymentCurrencies()->getPrimaryPaymentCurrencyIso();
     }
+
+    /**
+     * @param User $user
+     * @return void
+     * @throws MissingComponentException
+     * @throws Throwable
+     */
+    public function customerRegisteredHandler(User $user): void
+    {
+        $segments = Craft::$app->getRequest()->getActionSegments();
+        $userSaveSegments = ['users', 'save-user'];
+        $isUserSaveAction = $segments == $userSaveSegments;
+
+        // we have a cart number, currently anon, and the current action being executed is user save
+        if (!Craft::$app->getUser()->getIdentity() &&
```

Comment:
Do we need an extra check in here for if this is a new user? For example this will be called if a user saves their profile on the front end, does that have an impact on this code e.g. should it run?